File: rb/features.rb
# TheCat SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TheCatFeatures
  def self.make_feature(name)
    case name
    when "base"
      TheCatBaseFeature.new
    when "test"
      TheCatTestFeature.new
    else
      TheCatBaseFeature.new
    end
  end
end

File: py/features.py
# TheCat SDK feature factory

from feature.base_feature import TheCatBaseFeature
from feature.test_feature import TheCatTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TheCatBaseFeature(),
        "test": lambda: TheCatTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
